lib/Resque/Job.php

```diff
@@ -248,13 +252,15 @@ public function fail($exception)
 	 */
 	public function recreate()
 	{
-		$status = new Resque_Job_Status($this->payload['id'], $this->payload['prefix']);
 		$monitor = false;
-		if($status->isTracking()) {
-			$monitor = true;
+		if (isset($this->payload['id'])) {
+			$status = new Resque_Job_Status($this->payload['id'], $this->getPrefix());
+			if($status->isTracking()) {
+				$monitor = true;
+			}
 		}
 
-		return self::create($this->queue, $this->payload['class'], $this->getArguments(), $monitor, $this->payload['prefix']);
```

Comment:
Where does this new `null` come from?

Reply:
I noticed this when a test failed. After `$monitor` comes `$id`, and prefix was being passed. In this case we want a fresh `$id` and null is the default:
https://github.com/resque/php-resque/blob/7dd1474f73d5f2a4356cb2f69bd403476606c16e/lib/Resque/Job.php#L61

But this comment made me realise I should check for id being not empty, not just set, updated in 23fb4eab64fbafb51ebb7dfbe102aa5c13135098.